src/common/readfield_nc.f90

```diff
@@ -658,9 +660,186 @@ subroutine read_precipitation(ncid, nhdiff, timepos, timeposm1)
   where (precip < 0.0)
     precip = 0.0
   end where
+
+  if (requires_extra_precip_fields()) then
+    if ((met_params%mass_fraction_rain_in_air /= "") .and. &
+        (met_params%mass_fraction_cloud_condensed_water_in_air /= "")) then
+      call read_extra_precipitation_fields(ncid, timepos)
+    else if (met_params%mass_fraction_cloud_condensed_water_in_air /= "") then
+      call read_extra_precipitation_fields_infer_3d_precip(ncid, timepos)
+    else
+      error stop "Can not read extra 3D precipition for this meteorology"
+    endif
+  endif
+
+  call wetdep_precompute()
 end subroutine
 
 
+  subroutine read_extra_precipitation_fields(ncid, timepos)
+    use iso_fortran_env, only: error_unit, real64
+    use snaptabML, only: g
+    use snapfldML, only: ps2, precip3d, cw3d, cloud_cover, enspos
+    use snapgrdML, only: ahalf, bhalf, klevel
+    use snapdimML, only: nx, ny, nk
+    use snapmetML, only: met_params
+!> open netcdf file
+    integer, intent(in) :: ncid
+!> timestep in file
+    integer, intent(in) :: timepos
+    integer :: start(7), count(7)
+
+    real(real64), allocatable :: rain_in_air(:,:), graupel_in_air(:,:), snow_in_air(:,:)
```

Comment:
why are these fields 64bit floats? You will never get more than 3-4 significant digits.